### PR TITLE
Simplify and audit buttons

### LINKED ISSUE #397: Audit buttons patterns
## Context

Sanity check source code.

## Audit checklist

- [x] Browser testing
- [x] Accessibility testing
- [x] Sass linting
- [x] Update documentation
- [x] Convert sizing units to rems

### PR DESCRIPTION
**Pull request by barrymcgee**
## Done

- Add base styles for buttons
- Add button patterns as per spec in #397 

## QA

- Sanity check code
- Compare spec & button examples in `vf.io `

## Details

Fixes #397, #373, #354

### CODE REVIEW

File: scss/_base_button.scss [hunk 1]
```diff
@@ -0,0 +1,17 @@
+// Base button styles
+
+@mixin vf-b-button {
+
+  button {
+    font-family: $font-base-family;
```
Comment (anthonydillon) [inline scss/_base_button.scss]: Think you need to unquote this variable.

File: scss/_patterns_buttons.scss [hunk 2]
```diff
@@ -1,70 +1,53 @@
...
-// Primary button styling
-@mixin vf-primary-button {
-  button,
-  input[type='button'],
-  input[type='reset'],
-  input[type='submit'],
-  .button--primary {
-    @include vf-button($color-light, $color-light-grey, null);
+    &--neutral {
+      @extend .button;
```
Comment (anthonydillon) [inline scss/_patterns_buttons.scss]: Have you checked the CSS generated by this? Wonder if a placeholder would perform better.

**Comment by anthonydillon** [comment]
Looks good to me with a few questions. :+1:

File: scss/_patterns_buttons.scss (hunk 2 above)
Comment (barrymcgee) [inline scss/_patterns_buttons.scss]: Yes, compiled sass is as expected: http://www.sassmeister.com/gist/45bd0689c590c18597089cbc93597029

the reason why I didn't use a placehodler is becuase `.button` is the 'base' use-case which can then be modified to provide the other variants,

File: scss/_base_button.scss (hunk 1 above)
Comment (barrymcgee) [inline scss/_base_button.scss]: Done

**Pull request merged**